### ISSUE TITLE
Fails to download audio source with valid JSON request 

### ISSUE THREAD

**Issue by bryonye** [themoeway/yomitan#242]
**Description**
Using with Yomitan and custom URL (JSON) `http://localhost:8770/?expression={term}&term={term}` to get Chinese audio via the [Yomichan Forvo Server](https://ankiweb.net/shared/info/580654285) addon. I inspected the network traffic and the Forvo addon is generating a valid JSON with working links (validated against [this schema](https://github.com/themoeway/yomitan/blob/c4378974f1bea94947cbe6278b1d1593cfe653d9/ext/data/schemas/custom-audio-list-schema.json)), so the issue does not lie with the Forvo addon.

When adding a card, the Yomitan addon fails to add any audio; no file is created and the audio field is blank. When using the hover dictionary, there is an X by the speaker icon, and only a clicking sound plays, indicating no content. I think that perhaps the links are not being parsed properly [here](https://github.com/themoeway/yomitan/blob/c4378974f1bea94947cbe6278b1d1593cfe653d9/ext/js/media/audio-downloader.js#L229C28-L229C28). On Yomichan there is a similar issue where the actual JSON request is written to file instead of audio content, generating an invalid file. This may be related. 

**Browser version**
FireFox 117 on Linux

**Yomichan version**
Extension version: Yomitan 23.8.19.0
Platform: Linux
Browser: Firefox
Language: en-US
User agent: Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/117.0
AnkiConnect version: 6
Installed dictionaries: CEDICT

**Exported settings file**

**Comment by jbukl** [comment]
Tried replicating with different settings but it seemed to work fine. I had the same issue with the speaker button simply playing a click, but it was just because I had the permissions set incorrectly - for some reason it was defaulting to japaneseaudiopod101 until I enabled "all url access" as a permission, although I had already removed that audio source.

**Browser version**
Firefox Nightly 120.0a1 on Windows

**Yomichan version**
Extension version: Yomitan 23.10.1.0
Platform: Windows
Browser: Firefox
Language: en-US
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:120.0) Gecko/20100101 Firefox/120.0
AnkiConnect version: 6
Installed dictionaries: CEDICT

**Comment by praschke** [comment]
i experienced this on a fresh install, the reason was 'all url access' was not enabled. this is not a bug.

**Comment by bryonye** [comment]
This appears to be my bad then! It may be worth adding this issue to the FAQs, as the file permissions issue is only mentioned regarding Chrome. I went into `permissions.html` and found the 'Allow access to file URLs' setting and it (alongside every other option except for `clipboardRead`) is not toggle-able for me. I tried toggling `Enable recommended permissions` in `options.html`, and either way it's toggled I can't switch anything in the permissions page. Is there a way around this like manually editing a perms file?

**Comment by jbukl** [comment]
> This appears to be my bad then! It may be worth adding this issue to the FAQs, as the file permissions issue is only mentioned regarding Chrome. I went into `permissions.html` and found the 'Allow access to file URLs' setting and it (alongside every other option except for `clipboardRead`) is not toggle-able for me. I tried toggling `Enable recommended permissions` in `options.html`, and either way it's toggled I can't switch anything in the permissions page. Is there a way around this like manually editing a perms file?

Is this greyed out as well? The permission is actually not immediately on the permissions page, rather enabled on a popup from clicking this and then "Allow access to all urls"
![image](https://github.com/themoeway/yomitan/assets/148171160/e9977bb7-4f20-4543-9af8-dc267b14f440)

might make sense to have some sort of visual cue/note when you add audio sources that aren't reachable because of permissions.

**Comment by bryonye** [comment]
I can click the 'Configure allowed origins...' section, where it appears I already have 'Allow access to all URLs' enabled. 'Allow access to file URLs' is not toggleable. I thought it might be worth manually adding the localhost address to the 'Additional origins' list, but it appears that I am not able to actually add any URLs (or if it does add them, they are not listed anywhere). (Also, the URL parsing regex seems a bit odd, and could do with stripping whitespace at the start/end and perhaps not forcing trailing `/`. I am unsure if this kind of discussion warrants its own issue as it is still in pre-release.)

![image](https://github.com/themoeway/yomitan/assets/56762567/2bcbd1c7-e7a6-4a20-88e7-04ede0fc7d59)

![image](https://github.com/themoeway/yomitan/assets/56762567/c64963b4-aaa1-4ad7-86e3-80d3c3859151)

**Comment by praschke** [comment]
@bryonye on firefox those permissions can be enabled in the permissions tab of the extension in firefox's "manage extension" page, which can be accessed according to [this help page.](https://support.mozilla.org/en-US/kb/extensions-button)

**Comment by praschke** [comment]
permissions discoverability was addressed by #214, so it is likely that access to all urls is enabled and the allowed origins modal is not lying.

#173 also expressed this behavior with all urls enabled. when i looked into that there was a flakey bug where `declarativeNetRequest` would stop working, but only after the browser had restarted. i thought that flakiness explained the inconsistent behavior.

the version you're running, 23.8.19.0, includes that workaround for `declarativeNetRequest` by using the old `webRequest` path. this is probably a new bug.

**Comment by djahandarie** [comment]
@bryonye Could you confirm that you're still having an issue on the latest release?
